### PR TITLE
Adds /heartbeat url endpoint to webhook

### PR DESCRIPTION
**Pull request by petems**
* Useful to ensure server is up for loadbalancers, HA servers/proxies etc.

Rebase of #272 from @thebaron

* Params work taken out
* Commits squashed

### CODE REVIEW

File: spec/acceptance/signature_webhook_spec.rb [hunk 1]
```diff
@@ -47,5 +47,12 @@ class {'r10k::webhook':
         expect(r.exit_code).to eq(0)
       end
     end
+    it 'should respond with a 200 with status=success in JSON format on /heartbeat' do
```
Comment (ghoneycutt) [inline spec/acceptance/signature_webhook_spec.rb]: This does not actually check for a 200 HTTP status code. You could do that in a similar to fashion as done here with `curl -I -X GET`

Comment (petems) [inline spec/acceptance/signature_webhook_spec.rb]: Fixed 👍

**Comment by petems** [comment]
Ironically, all the acceptance tests except the hearbeat ones are failing 😓 

```
Finished in 5 minutes 13 seconds (files took 40.01 seconds to load)
24 examples, 8 failures

Failed examples:

rspec ./spec/acceptance/basic_webhook_spec.rb:43 # System Ruby with No SSL, Not protected, No mcollective default parameters should support style Github payloads via payload end point
rspec ./spec/acceptance/basic_webhook_spec.rb:49 # System Ruby with No SSL, Not protected, No mcollective default parameters should support style Gitorious payloads via payload end point
rspec ./spec/acceptance/basic_webhook_spec.rb:55 # System Ruby with No SSL, Not protected, No mcollective default parameters should support style BitBucket payloads via payload end point
rspec ./spec/acceptance/prefix_webhook_spec.rb:61 # Prefix Enabled,System Ruby with No SSL, Not protected, No mcollective default parameters should calculate webteam prefix Github payloads via payload end point
rspec ./spec/acceptance/prefix_webhook_spec.rb:67 # Prefix Enabled,System Ruby with No SSL, Not protected, No mcollective default parameters should calculate secteam prefix with Github payloads via payload end point
rspec ./spec/acceptance/prefix_webhook_spec.rb:73 # Prefix Enabled,System Ruby with No SSL, Not protected, No mcollective default parameters should calculate custom prefix with Github payloads via payload end point
rspec ./spec/acceptance/prefix_webhook_spec.rb:80 # Prefix Enabled,System Ruby with No SSL, Not protected, No mcollective default parameters should calculate no prefix Github payloads via payload end point
rspec ./spec/acceptance/signature_webhook_spec.rb:41 # GitHub Secret Enabled, System Ruby with No SSL, Not protected, No mcollective default parameters should support style Github payloads via payload end point with signature in header
```

Looks like a permission error:

```
 8) GitHub Secret Enabled, System Ruby with No SSL, Not protected, No mcollective default parameters should support style Github payloads via payload end point with signature in header
     Failure/Error: expect(r.stdout).to match(/^.*success.*$/)
       expected "{\"status\":\"fail\",\"message\":\"\\nERROR\\t -> Unable to determine current branches for Git sourc.../httpserver.rb:94:in `run'\",\"/usr/share/ruby/webrick/server.rb:295:in `block in start_thread'\"]}" to match /^.*success.*$/
       Diff:
       @@ -1,2 +1,2 @@
       -/^.*success.*$/
       +"{\"status\":\"fail\",\"message\":\"\\nERROR\\t -> Unable to determine current branches for Git source 'puppet' (/etc/puppet/environments)\\nOriginal exception:\\nPermission denied - /var/cache/r10k\\n\",\"trace\":[\"/usr/local/bin/webhook:193:in `run_command'\",\"/usr/local/bin/webhook:232:in `deploy'\",\"/usr/local/bin/webhook:159:in `block in <class:Server>'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:1611:in `call'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:1611:in `block in compile!'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:975:in `[]'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:975:in `block (3 levels) in route!'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:994:in `route_eval'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:975:in `block (2 levels) in route!'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:1015:in `block in process_route'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:1013:in `catch'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:1013:in `process_route'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:973:in `block in route!'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:972:in `each'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:972:in `route!'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:1085:in `block in dispatch!'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:1067:in `block in invoke'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:1067:in `catch'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:1067:in `invoke'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:1082:in `dispatch!'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:907:in `block in call!'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:1067:in `block in invoke'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:1067:in `catch'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:1067:in `invoke'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:907:in `call!'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:895:in `call'\",\"/usr/local/share/gems/gems/rack-protection-1.5.3/lib/rack/protection/xss_header.rb:18:in `call'\",\"/usr/local/share/gems/gems/rack-protection-1.5.3/lib/rack/protection/path_traversal.rb:16:in `call'\",\"/usr/local/share/gems/gems/rack-protection-1.5.3/lib/rack/protection/json_csrf.rb:18:in `call'\",\"/usr/local/share/gems/gems/rack-protection-1.5.3/lib/rack/protection/base.rb:49:in `call'\",\"/usr/local/share/gems/gems/rack-protection-1.5.3/lib/rack/protection/base.rb:49:in `call'\",\"/usr/local/share/gems/gems/rack-protection-1.5.3/lib/rack/protection/frame_options.rb:31:in `call'\",\"/usr/local/share/gems/gems/rack-1.6.5/lib/rack/nulllogger.rb:9:in `call'\",\"/usr/local/share/gems/gems/rack-1.6.5/lib/rack/head.rb:13:in `call'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/show_exceptions.rb:25:in `call'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:182:in `call'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:2013:in `call'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:1487:in `block in call'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:1787:in `synchronize'\",\"/usr/local/share/gems/gems/sinatra-1.4.8/lib/sinatra/base.rb:1487:in `call'\",\"/usr/local/share/gems/gems/rack-1.6.5/lib/rack/handler/webrick.rb:88:in `service'\",\"/usr/share/ruby/webrick/httpserver.rb:138:in `service'\",\"/usr/share/ruby/webrick/httpserver.rb:94:in `run'\",\"/usr/share/ruby/webrick/server.rb:295:in `block in start_thread'\"]}"
```

**Comment by dhollinger** [comment]
@petems closing as #272 was merged last week

**Pull request closed without merging**